src/login/components/LoginFormLedger/LoginFormLedger.js

```diff
@@ -56,50 +66,63 @@ export default class LoginFormLedger extends React.Component {
   }
 
   renderActions = () => {
-    const disabled = this.props.disabled || !this.props.deviceInfo;
+    const disabled = (this.state.connected && this.state.open) ? this.props.disabled : true;
     const onClick = disabled ? null : this.handleLogin;
 
+    this.state.deviceError = (!disabled) ?
+      null :
+      this.state.deviceError;
+
     return (
       <div className={styles.actions}>
         <Button type="button" disabled={disabled} onClick={onClick}>Login</Button>
       </div>
     );
   }
 
-  renderDeviceInfo = () => {
-    const { deviceInfo } = this.props;
+  handleLogin = (event) => {
+    const { publicKey, onLogin } = this.props;
 
-    if (deviceInfo) {
-      return <p>Found USB {deviceInfo.manufacturer} {deviceInfo.product}</p>;
-    } else {
-      return <p>Searching for USB devices. Please plug in your Ledger to login.</p>;
-    }
+    event.preventDefault();
+    onLogin({ publicKey });
   }
 
-  renderDeviceStatus = () => {
-    const { publicKey } = this.props;
+  getDeviceInfo = () => {
```

Comment:
For consistency & clarity, I would prefer that we ensure any function that returns React nodes we should prefix with `render` rather than `get`.

I also think this function (along with `getDeviceStatus` below) is overloaded in what it's doing.  It's attempting to inspect device status, change state, and render the result.  I think it would be easier to follow what's happening if these things remain separate.